DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/DelegateProgress/LearningLog.cshtml

```diff
@@ -23,7 +23,7 @@
   }
 }
 
-<div class="nhsuk-grid-row">
+<div class="nhsuk-grid-row" id="@(Model.JavascriptSearchSortFilterPaginateEnabled ? "js-styling-hidden-area-while-loading" : "no-js-styling")">

```

Comment:
I believe the addition of this class will cause its contents to be hidden on page load until the JS reveals it (which might take a few seconds if there are a significant number of records). We should add the loading spinner partial above this (as we do on other search/sort/filter/paginate pages, such as All Delegates) so that it is more obvious to the user that something is happening in the background.

This has also highlighted another problem that wasn't obvious before - the JS will error when there are no learning log entries. When there are no entries, we replace the table/controls with a simple message, but that prevents the JS from finding elements it needs. Now, this means that page content will never be revealed, as the JS breaks before it can reveal it. So that might require a little further refactoring of either the page or the JS.

Reply:
Fixed both issues